Automattic/wp-calypso
# Subscription Management: Record tracks event when changing subscription settings.

Fixes https://github.com/Automattic/wp-calypso/issues/78290

## Proposed Changes

* This PR records the following tracks event:
  * calypso_subscriptions_notifications_toggle_on
  * calypso_subscriptions_notifications_toggle_off
  * calypso_subscriptions_post_emails_toggle_on
  * calypso_subscriptions_post_emails_toggle_off
  * calypso_subscriptions_comment_emails_toggle_on
  * calypso_subscriptions_comment_emails_toggle_off
  * calypso_subscriptions_post_emails_set_frequency

* They are the Subscriptions equivalent of the Reader's `calypso_reader_*` tracks event.

* This PR introduces `SubscriptionManagerContext` which I used for passing configs, right now it has only one property, `portal`.

## Testing Instructions

* Go to `/subscriptions/sites`.
* Open Network panel and filter by `.gif`.
* Try changing all subscription settings in the settings popover.
* It should record tracks event.
* Repeat the same at `/read/subscriptions`.
* It should record tracks event with a different portal value.

<img width="1094" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1287077/b58247c2-d5b6-43d2-9655-81cdb1ffb975">

## Pre-merge Checklist
